quipucords/scanner/ansible/inspect.py

```diff
@@ -96,6 +106,32 @@ def _init_connection_stats(self):
             sys_unreachable=0,
         )
 
+    def _detect_ansible_endpoints(self):
+        """
+        Detect the Ansible endpoints.
+
+        This is called to determine the Ansible endpoints we need for the Ansible
+        Automation Platform controller we are connected to. Handles both types:
+        - Controller type endpoint (Ansible <=2.4 - i.e., port 8443)
+        - Gateway type endpoint (Ansible >=2.5 - i.e., port 443) which results
+        in /api/controller/v# type endpoints.
+        """
+        if self.endpoints:
+            return
+        ctrl_ep = self.client.get("/api/", **self.REQUEST_KWARGS).json()
+        if ctrl_ep.get("apis"):
+            ctrl_url = ctrl_ep.get("apis")["controller"]
+            ctrl_ep = self.client.get(ctrl_url, **self.REQUEST_KWARGS).json()
+        v2_ctrl_url = ctrl_ep.get("available_versions")["v2"]
+        v2_ctrl_ep = self.client.get(v2_ctrl_url, **self.REQUEST_KWARGS).json()
+
+        self.endpoints = self.AAP_ENDPOINTS(
```

Comment:
Should we fail louder if future AAP breaks backwards compatibility and any of these keys is no longer provided?

Maybe `self.client.get(None)` is going to be caught by some generic error handler elsewhere, I don't know.